app/src/org/commcare/views/EntityView.java

```diff
@@ -167,7 +167,7 @@ public void setSearchTerms(String[] terms) {
     }
 
     public void refreshViewsForNewEntity(Entity e, boolean currentlySelected, long rowId) {
-        for (int i = 0; i < e.getNumFields(); ++i) {
+        for (int i = 0; i < views.length; ++i) {

```

Comment:
They can be used interchangeably because `views` is setup like this:
```
this.views = new View[e.getNumFields()];
```
I'm just trying to make the internal representation have as few external dependencies as possible.